pkg/interfaces/contracts/standalone-utils/IHyperEVMRateProviderFactory.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { IHyperEVMRateProvider } from "./IHyperEVMRateProvider.sol";
+
+interface IHyperEVMRateProviderFactory {
+    /**
+     * @notice A new HyperEVM Rate Provider was created.
+     * @param tokenIndex The index of the base asset on Hyperliquid API
+     * @param pairIndex The pair index of the spot price on Hyperliquid API
+     * @param rateProvider The address of the deployed rate provider
+     */
+    event RateProviderCreated(
+        uint256 indexed tokenIndex,
+        uint256 indexed pairIndex,
+        IHyperEVMRateProvider rateProvider
+    );
+
+    /// @notice Emitted when the factory is disabled.
+    event RateProviderFactoryDisabled();
+
+    /**
+     * @notice A rate provider already exists for the given token and pair.
+     * @param tokenIndex The index of the base asset on Hyperliquid API
+     * @param pairIndex The pair index of the spot price on Hyperliquid API
+     * @param rateProvider The address of the deployed rate provider
+     */
+    error RateProviderAlreadyExists(uint32 tokenIndex, uint32 pairIndex, IHyperEVMRateProvider rateProvider);
+
+    /// @notice The factory is disabled.
+    error RateProviderFactoryIsDisabled();
+
+    /**
+     * @notice Returns a number representing the rate provider version.
+     * @return rateProviderVersion The rate provider version number
+     */
+    function getRateProviderVersion() external view returns (uint256 rateProviderVersion);
+
+    /**
+     * @notice Creates a new HyperEVM Rate Provider for the given token and pair.
+     * @param tokenIndex The index of the base asset on Hyperliquid API
+     * @param pairIndex The pair index of the spot price on Hyperliquid API
+     * @return rateProvider The address of the deployed rate provider
+     */
+    function create(uint32 tokenIndex, uint32 pairIndex) external returns (IHyperEVMRateProvider rateProvider);
+
+    /**
+     * @notice Gets the rate provider for the given token and pair.
```

Comment:
```suggestion
     * @notice Gets the rate provider for the given token and pair.
     * @dev Returns the zero address if not found.
```
Maybe that's obvious? But it could have also been written to revert, so maybe say what it does here.